x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/engines/fetch_engine_api_logic.ts

```diff
@@ -5,22 +5,22 @@
  * 2.0.
  */
 
-import { EnterpriseSearchEngineDetails } from '../../../../../common/types/engines';
+import { EnterpriseSearchEngine } from '../../../../../common/types/engines';
 import { Actions, createApiLogic } from '../../../shared/api_logic/create_api_logic';
 import { HttpLogic } from '../../../shared/http';
 
 export interface FetchEngineApiParams {
   engineName: string;
 }
 
-export type FetchEngineApiResponse = EnterpriseSearchEngineDetails;
+export type FetchEngineApiResponse = EnterpriseSearchEngine;
```

Comment:
These two types are different and this change will break things.

`EnterpriseSearchEngine` is returned from `GET /api/engines` list endpoint
`EnterpriseSearchEngineDetails` is returned from `GET /api/engines/<engine_name>` endpoint

Ensure your `enterprise_search` is updated if your still getting `EnterpriseSearchEngine` from the get engine endpoint.

Reply:
I'm open to a better name for `EnterpriseSearchEngineDetails`, just need something to distinguish the list results from the get for a specific engine.

Reply:
oh yes they are different. What was I thinking 🤦

Reply:
quick question: why is there a difference between the two? Is it for performance reasons?

Reply:
@sphilipse exactly performance reasons. When you get a single engine we hydrate information about it's indices, similar to `fetch_indices` in Kibana. This is so that we can display Document count, ingestion method etc.

But when listing all engines that would add latency and N+1 queries since we would then have to get `_stats` for every index that is a part of every engine on that page of results. For example this is why we have for now dropped the document count from the engine list page that was a part of the initial design.